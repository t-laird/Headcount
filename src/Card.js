import React from 'react';

function Card (props) {
  const mappedData = Object.keys(props.data).map((dataPoint, index) => {
    return (
      <li key={`li-${index}`}>{dataPoint}: {props.data[dataPoint]}</li>
    )
  })

  return (
    <div className="Card">
      <h3>{props.district}</h3>
      <ul>
        {mappedData}
      </ul>
    </div>
  );
}

export default Card;