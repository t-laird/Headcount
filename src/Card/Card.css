.Card {
  border: none;
  border-radius: 10px;
  box-shadow: 5px 5px 5px 5px darkgrey;
  margin: 15px;
  padding: 0px;
  width: 200px;
  min-width: 200px;
  box-sizing: border-box;
  display: flex;
  cursor: pointer; }
  .Card div {
    padding: 15px; }
  .Card h4 {
    height: 2em;
    margin: 0px;
    text-align: center;
    width: 100%; }
  .Card ul {
    margin: 15px 0 0;
    padding: 3px; }
    .Card ul li {
      list-style: none; }
      .Card ul li p {
        display: inline-block;
        width: 40px;
        margin: 0; }
    .Card ul .high {
      color: blue; }
    .Card ul .not-yet {
      color: lightblue; }
  .Card:hover {
    box-shadow: 5px 5px 5px 5px black; }

.selected {
  background-color: #ffcccc; }

.Card.Comparison {
  border: 1px dashed black;
  background-color: lightgrey;
  height: 270px; }

.Card.compared {
  border: 1px dashed black;
  text-align: center; }
