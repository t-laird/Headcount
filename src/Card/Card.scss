@import '../variable.scss';
@import '../../fontello-e98b1dbc/css/fontello.css';

.Card {
  display: flex;

  border: none;
  border-radius: 10px;
  box-shadow: $card-box-shadow;
  box-sizing: border-box;
  
  margin: 15px;
  padding: 0px;
  min-width: 200px;
  width: 200px;

  cursor: pointer;
  transition: 1s;

  &:hover {
    box-shadow: $card-hover-box-shadow;  
  }

  div {
    width: 100%;
  }

  h4 {
    display: flex;
    align-items: center;
    justify-content: center;
    text-align: center;
    
    background: $main-color;
    border-radius: 10px 10px 0 0;

    height: 3em;
    margin: 0px;
    padding: 3px;
  }

  ul {
    margin: 10px 0 0;
    padding: 10px;

    li {
      display: flex;
      justify-content: space-around;
      list-style: none;
      p {
        margin: 0;
      }
      p.data {
        text-align: right;
        width: 40px;
      }
      i.low{
        color: $low-color;
      }
      i.low-average{
        color: $low-average-color;
      }
      i.average{
        color: $average-color;
      }
      i.high-average{
        color: $high-average-color;
      }
      i.high{
        color: $high-color;
      }
    }
  }
}

.selected {
  background-color: $card-selected-background;
}

.Card.comparison {
  background-color: $comparison-empty-background;
  border: $dashed-border;
}

.Card.compared {
  border: $dashed-border;
  text-align: center;
}

.Card.displayComparedData {
  display: flex;
  
  div {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;

    border-radius: 10px;
    height: 100%;
    
    h3 {
      display: flex;
      align-items: center;
      justify-content: center;

      background: $main-color;
      border-radius: 10px 10px 0 0;
      color: black;
      font-size: 14pt;
      
      margin: 0px;
      padding-top: 6px;
      height: 3em;
      i {
        color: $graph-button-color;
        font-size: 30px;
      }
      &:nth-child(3) {
        background:  $card-selected-background;
        border-radius: 0px;
        border-bottom: $solid-border;
        border-top: $solid-border;

        margin: 0px;
      }
      &:nth-child(5) {
        border-radius: 0 0 10px 10px;
      }
    }

    p {
      margin: 0px;
    }
    i.low{
      color: $low-color;
    }
    i.low-average{
      color: $low-average-color;
    }
    i.average{
      color: $average-color;
    }
    i.high-average{
      color: $high-average-color;
    }
    i.high{
      color: $high-color;
    }
  }
}