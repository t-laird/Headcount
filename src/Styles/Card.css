.Card {
  border: none;
  border-radius: 10px;
  box-shadow: 5px 5px 5px 5px darkgrey;
  margin: 15px;
  padding: 15px;
  width: 200px;
  box-sizing: border-box; }
  .Card h4 {
    height: 2em; }
  .Card ul {
    margin: 0px;
    padding: 3px; }
    .Card ul li {
      list-style: none;
      width: 200px; }

.Card.Comparison {
  border: 1px dashed black; }
