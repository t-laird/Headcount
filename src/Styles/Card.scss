.Card {
  border: none;
  border-radius: 10px;
  box-shadow: 5px 5px 5px 5px darkgrey;
  margin: 15px;
  padding: 15px;
  width: 200px;
  box-sizing: border-box;
  h4 {
    height: 2em;
  }
  ul {
    margin: 0px;
    padding: 3px;
    li {
      list-style: none;
      width: 200px;
    }
  }
}

.Card.Comparison {
  border: 1px dashed black;
}