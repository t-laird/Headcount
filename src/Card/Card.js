import React from 'react';
import PropTypes from 'prop-types';
import './Card.css';

const Card = (props) => {
  const mappedData = Object.keys(props.data).map((dataPoint, index) => {
    const liStyle = props.data[dataPoint] > 0.5 ? 'high' : 'not-yet';
    return (
      <li key={`li-${index}`} className={liStyle}>
        <p>{dataPoint}:</p> {props.data[dataPoint]}
      </li>
    );
  });

  return (
    <div className={props.type}>
      {
        props.district &&
        <div className='card-content' onClick={() => {props.selectCard(props.district);}}>
          <h4>{props.district}</h4>
          <ul>
            {mappedData}
          </ul>
        </div>
      }
      {
        (props.type === "Card compared") &&
          <div>
            <h1>Click Two Cards to Compare Them</h1>
          </div>
      }
      {
        (props.type === "Card displayComparedData") &&
        <div>
          <h3> {props.district1}: {props.compareData[props.district1]}</h3>
          <h3> {props.compareData.compared}</h3>
          <h3> {props.district2}: {props.compareData[props.district2]}</h3>
        </div>
      }
    </div>
  );
};

Card.propTypes = {
  type: PropTypes.string,
  data: PropTypes.object,
  selectCard: PropTypes.func,
  district: PropTypes.string,
  district1: PropTypes.string,
  district2: PropTypes.string,
  compareData: PropTypes.object
}

export default Card;