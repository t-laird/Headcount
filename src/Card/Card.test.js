import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import { shallow, mount } from 'enzyme';

describe('Card test', () => {
  it('should render the correct', () => {
    const renderedCard = shallow(
      <Card 
        type='Card' 
        district="Colorado" 
        data={{}}/>);

    expect(renderedCard.find('h4').length).toEqual(1);
    expect(renderedCard.find('ul').length).toEqual(1);
    expect(renderedCard.find('.Card').length).toEqual(1);
  });

  it('should match the Card snapshot', () => {
    const renderedCard = shallow(
      <Card 
        type='Card' 
        data={{}}/>);

    expect(renderedCard).toMatchSnapshot();
  });

  it('should render the correct', () => {
    const renderedCard = shallow(
      <Card 
        type='Card compared' 
        data={{}}/>);

    expect(renderedCard.find('h1').length).toEqual(1);
    expect(renderedCard.find('h4').length).toEqual(0);
    expect(renderedCard.find('ul').length).toEqual(0);
  });

  it('should render the correct', () => {
    const renderedCard = shallow(
      <Card type='Card displayComparedData'
        data={{}}
        district1="Colorado" 
        district2='ADAMS COUNTY 14' 
        compareData={true}/>);

    expect(renderedCard.find('h3').length).toEqual(3);
    expect(renderedCard.find('h4').length).toEqual(0);
    expect(renderedCard.find('ul').length).toEqual(0);
  });

  it('should call the method passed from props in the onClick', () => {
    const mockFunc = jest.fn();

    const renderedCard = shallow(
      <Card 
        type='Card' 
        district="Colorado" 
        selectCard={mockFunc} data={{}}/>);

    const cardDiv = renderedCard.find('.card-content');

    cardDiv.simulate('click');
    expect(mockFunc.mock.calls.length).toEqual(1);
  });
});
