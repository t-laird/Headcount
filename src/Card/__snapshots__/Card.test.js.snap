exports[`Card test should match the Card snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={Array []}
    type="Card" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        false,
        false,
      ],
      "className": "Card",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      false,
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          false,
          false,
        ],
        "className": "Card",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        false,
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
